lib/puppet/provider/package/pip.rb

```diff
@@ -56,11 +56,8 @@ def query
   # cache of PyPI's package list so this operation will always have to
   # ask the web service.
   def latest
-    client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
-    client.http_header_extra = {"Content-Type" => "text/xml"}
-    client.timeout = 10
-    result = client.call("package_releases", @resource[:name])
-    result.first
+    result = JSON.parse(open("https://pypi.python.org/pypi/#{@resource[:name]}/json").read)
```

Comment:
The third option is to use net::http directly (without puppet's HTTP pools) and pass in the values from the `puppet::util::http_proxy` class, just as `network::http::factory` does. That avoids fighting with puppet's CA mess (at the expense of some duplicate code).

But realistically we already have several providers that need to convert the proxy info into new formats for the programs they execute, so it doesn't seem that evil to do it here.